### PR TITLE
Added a toggle for bounding boxes and segmentation

### PR DESCRIPTION
**Pull request by troydiaz**
## Description
Briefly describe the purpose of this pull request.

## Related Issue
Link to any related GitHub issue(s).

## Testing
Describe how you tested your changes (e.g., screenshots, logs, build output).

## Checklist
- [ ] Code builds and runs locally
- [ ] Feature branch follows naming convention (feature/<description>)
- [ ] Changes reviewed by at least one teammate
- [ ] Documentation updated (if needed)
- [ ] No sensitive data or credentials committed

### CODE REVIEW

File: models/yolo_people_fire_smoke/main.py [hunk 1]
```diff
@@ -41,69 +74,82 @@
...
     if dt > 0:
-        fps_hist.append(1 / dt)
+        fps_hist.append(1.0 / dt)
+
+    # --- Inference ---
+    # People: use seg model when SEG_ON, else det model
+    if SEG_ON:
+        people_results = people_seg_model.predict(frame, conf=0.4, classes=[0], verbose=False)
+    else:
+        people_results = people_det_model.predict(frame, conf=0.4, classes=[0], verbose=False)
```
Comment (Kris-09) [inline models/yolo_people_fire_smoke/main.py]: When `SEG_ON` is true, `people_seg_model.predict()` returns masks and boxes, but the code still assumes `people_results[0].boxes.cls == 0`. Some segmentation models return masks even when a detection is extremely low confidence or partially missing. So, it might be safer to add a defensive check.

`if people_results and people_results[0].boxes is not None:`

Reference: https://docs.ultralytics.com/reference/engine/results/

File: models/yolo_people_fire_smoke/main.py
Comment (Kris-09) [inline models/yolo_people_fire_smoke/main.py]: The people counting logic behaves differently when `SEG_ON` is true vs false, but there are no tests to verify consistency between detection based and segmentation based counting methods. We can add unit tests that verify people counting accuracy in both modes, especially for edge cases like low confidence detections and empty results.

Reference: https://docs.python.org/3/library/unittest.html
